src/karts/controller/skidding_ai.cpp

```diff
@@ -1172,80 +1192,252 @@ void SkiddingAI::handleItems(const float dt)
         }
         return;
     }
-
-    // Tactic 1: wait ten seconds, then use item
+   
+   int ai_skill = 0;
+   if (m_ai_properties->m_item_usage_skill > 0)
+   {
+      if (m_ai_properties->m_item_usage_skill > 5)
+      {
+         ai_skill = 5;
+      }
+      else
+      {
+         ai_skill = m_ai_properties->m_item_usage_skill;
+      }
+   }
+   
+   if (m_kart->getBoostAI() == true)
+   {
+      if (ai_skill < 5)
+      {
+         ai_skill = ai_skill + 1; //possible improvement : make the boost amplitude pulled from config
+      }
+   }
+
+    // Tactic 0: don't use item
     // -----------------------------------------
-    if(!m_ai_properties->m_item_usage_non_random)
+    if(ai_skill == 0)
+    {
+        return;
+    }
+
+    // Tactic 1: wait between 5 and 10 seconds, then use item
+    // ------------------------------------------------------
+    if(ai_skill == 1)
     {
-        if( m_time_since_last_shot > 10.0f )
+       int random_t = 0;
+       RandomGenerator g = RandomGenerator();
```

Comment:
Don't create a new one everytime, the way RandomGenerator is currently implemented this will not end well